src/neo-asm/ASM_Language/dasm/DASM.cs

```diff
@@ -0,0 +1,338 @@
+                            jmpaddrs.Add(jmpaddr);
+                    }
+                }
+            }
+            jmpaddrs.Sort();
+            //gen code
+            for (var i = 0; i < codes.Count; i++)
+            {
+                var code = codes[i];
+                if (code.addr >= range.addrbegin && (range.addrend < 0 || code.addr < range.addrend))
+                {//in code range
+                    if (jmpaddrs.Contains(code.addr))
+                    {
+                        ASMLabel label = new ASMLabel();
+                        label.label = "label" + (jmpaddrs.IndexOf(code.addr) + 1);
+                        label.commentRight = "//addr:" + code.addr;
+                        func.nodes.Add(label);
+                    }
+                    ASMInstruction inst = new ASMInstruction();
+                    if (code.isPush)
+                    {
+                        inst.opcode = ASMOpCode.CreatePush();
+                        if (code.opcode == VM.OpCode.PUSH0)
+                        {
+                            inst.valuetext = "0";
+                            inst.commentRight = "// or false or empty";
+                        }
+                        else if (code.opcode == VM.OpCode.PUSHM1)
+                        {
+                            inst.valuetext = "-1";
+                        }
+                        else if (code.opcode == VM.OpCode.PUSH1)
+                        {
+                            inst.valuetext = "1";
+                            inst.commentRight = "// or true";
+                        }
+                        else
+                        {
+                            inst.valuetext = Data2Str(code.data);
+                            inst.commentRight = Data2Comment(code.data);
+                        }
+                    }
+                    else
+                    {
+                        inst.opcode = ASMOpCode.Create(code.opcode);
+                        if (code.opcode == VM.OpCode.CALL)
+                        {
+                            var calladdr = code.addr + code.offset;
+                            var callname = mapfuncrange[calladdr].name;
+                            inst.valuetext = callname;
+                        }
+                        else if (code.opcode == VM.OpCode.JMP || code.opcode == VM.OpCode.JMPIF || code.opcode == VM.OpCode.JMPIFNOT)
+                        {
+                            var jmpaddr = code.addr + code.offset;
+                            var label = "label" + (jmpaddrs.IndexOf(jmpaddr) + 1);
+                            inst.valuetext = label;
+                        }
+                        else if (code.opcode == VM.OpCode.SYSCALL)
+                        {
+                            inst.valuetext = Data2Str(code.data);
+                        }
+                        else
+                        {
+                            //no param
+                        }
```

Comment:
I'm not a big fan of an empty else. Do you need this for some particular reason?